src/test/java/solution/EnumBoardTest.java

```diff
@@ -0,0 +1,282 @@
+package solution;
+
+import java.awt.Point;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import solution.Solution;
+import solution.BacktrackingSolution;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class EnumBoardTest {
+
+    private char[][] unsolvedArray = new char[9][9];
+    private SudokuBoard unsolvedBoard;
+
+    @BeforeEach
+    void setUp() {
+        unsolvedArray[0] = new char[]{'5', '3', '.', '.', '7', '.', '.', '.', '.'};
+        unsolvedArray[1] = new char[]{'6', '.', '.', '1', '9', '5', '.', '.', '.'};
+        unsolvedArray[2] = new char[]{'.', '9', '8', '.', '.', '.', '.', '6', '.'};
+        unsolvedArray[3] = new char[]{'8', '.', '.', '.', '6', '.', '.', '.', '3'};
+        unsolvedArray[4] = new char[]{'4', '.', '.', '8', '.', '3', '.', '.', '1'};
+        unsolvedArray[5] = new char[]{'7', '.', '.', '.', '2', '.', '.', '.', '6'};
+        unsolvedArray[6] = new char[]{'.', '6', '.', '.', '.', '.', '2', '8', '.'};
+        unsolvedArray[7] = new char[]{'.', '.', '.', '4', '1', '9', '.', '.', '5'};
+        unsolvedArray[8] = new char[]{'.', '.', '.', '.', '8', '.', '.', '7', '9'};
+
+        unsolvedBoard = new EnumBoard(unsolvedArray);
+    }
+
+    @Test
+    void Should_ThrowIllegalArgumentException_When_ArrayDimensions0x0() {
+        char[][] empty = new char[0][0];
+        Exception thrown =
```

Comment:
This formatting is weird; it unnecessarily spans over too many lines. Unfortunately, the KIE conventions won't fix it completely (or they might have actually cause it).

If the lambda is a one-liner, no need to use curly braces.

The `thrown` variable can be probably omitted, `AssertJ` provides a fluent API, that lets you to chain individual commands.

Reply:
Now I realized that you are not using AssertJ, but the jUnit5 assertions.